fluence-faas/src/raw_toml_config.rs

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright 2020 Fluence Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+use crate::FaaSError;
+use crate::Result;
+use crate::config::*;
+
+use serde_derive::Serialize;
+use serde_derive::Deserialize;
+
+use std::convert::TryInto;
+use std::collections::HashMap;
+use std::collections::HashSet;
+use std::path::PathBuf;
+
+/*
+An example of the config:
+
+modules_dir = "wasm/artifacts/wasm_modules"
+
+[[module]]
+    name = "ipfs_node.wasm"
+    mem_pages_count = 100
+    logger_enabled = true
+
+    [module.imports]
+    mysql = "/usr/bin/mysql"
+    ipfs = "/usr/local/bin/ipfs"
+
+    [module.wasi]
+    envs = { "IPFS_ADDR" = "/dns4/relay02.fluence.dev/tcp/15001" }
+    preopened_files = ["/Users/user/tmp"]
+    mapped_dirs = {"tmp" = "/Users/user/tmp"}
+
+[default]
+    mem_pages_count = 100
+    logger_enabled = true
+
+    [default.imports]
+    mysql = "/usr/bin/mysql"
+    ipfs = "/usr/local/bin/ipfs"
+
+    [default.wasi]
+    envs = []
+    preopened_files = ["/Users/user/tmp"]
+    mapped_dirs = {"tmp" = "/Users/user/tmp"}
+ */
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct TomlFaaSConfig {
+    pub modules_dir: Option<String>,
+    #[serde(skip_serializing_if = "Vec::is_empty")]
+    pub module: Vec<TomlFaaSNamedModuleConfig>,
+    pub default: Option<TomlFaaSModuleConfig>,
+}
+
+impl TomlFaaSConfig {
+    /// Load config from filesystem.
+    pub fn load<P: Into<PathBuf>>(path: P) -> Result<Self> {
+        let path = path.into();
+        let file_content = std::fs::read(&path)?;
+        toml::from_slice(&file_content).map_err(|e| {
+            FaaSError::ConfigParseError(format!("Error parsing config {:?}: {:?}", path, e))
+        })
+    }
+}
+
+impl TryInto<FaaSConfig> for TomlFaaSConfig {
+    type Error = FaaSError;
+
+    fn try_into(self) -> Result<FaaSConfig> {
+        from_toml_faas_config(self)
+    }
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, Default)]
+pub struct TomlFaaSNamedModuleConfig {
```

Comment:
Let's call it `TomlFaaSModuleConfig`, so it is consistent with `FaaSModuleConfig`